src/Ops.Service/FeatureService.cs

```diff
@@ -0,0 +1,116 @@
+        {
+            _logger = logger
+                ?? throw new ArgumentNullException(nameof(logger));
+            _featureRepository = featureRepository
+                ?? throw new ArgumentNullException(nameof(featureRepository));
+        }
+
+        public async Task<DataWithCount<ICollection<Feature>>> GetPaginatedListAsync(
+            BaseFilter filter)
+        {
+            return await _featureRepository.GetPaginatedListAsync(filter);
+        }
+
+        public async Task<Feature> GetFeatureByNameAsync(string featureName)
+        {
+            try
+            {
+                return await _featureRepository.GetFeatureByName(featureName);
+            }
+            catch (OcudaException ex)
+            {
+                _logger.LogError(ex.Message);
+                throw new OcudaException($"The feature: " + featureName + " was unable to be found.");
+            }
+        }
+
+        public async Task<Feature> AddFeatureAsync(Feature feature)
+        {
+            try
+            {
+                feature.Name = feature.Name?.Trim();
+                await ValidateAsync(feature);
+
+                await _featureRepository.AddAsync(feature);
+                await _featureRepository.SaveAsync();
+            }
+            catch (OcudaException ex)
+            {
+                _logger.LogError(ex.Message);
+                throw new OcudaException(ex.Message);
+            }
+
+            return feature;
+        }
+
+        public async Task<Feature> EditAsync(Feature feature)
+        {
+            var currentFeature = await _featureRepository.FindAsync(feature.Id);
+            if(currentFeature != null)
+            {
+                try
+                {
+                    currentFeature.BodyText = feature.BodyText;
+                    currentFeature.FontAwesome = feature.FontAwesome;
+                    currentFeature.Name = feature.Name;
+                    currentFeature.Stub = feature.Stub;
+                    _featureRepository.Update(feature);
+                    await _featureRepository.SaveAsync();
+                    return await _featureRepository.FindAsync(currentFeature.Id);
+                }
+                catch (OcudaException ex)
+                {
+                    _logger.LogError(ex.Message);
+                    throw new OcudaException(ex.Message);
+                }
+            }
+            else
+            {
+                throw new OcudaException($"Could not find Feature.");
+            }
+        }
+
+        public async Task DeleteAsync(int id)
+        {
+            try
+            {
+                _featureRepository.Remove(id);
+                await _featureRepository.SaveAsync();
+            }
+            catch (OcudaException ex)
+            {
+                _logger.LogError(ex.Message);
+                throw new OcudaException(ex.Message);
+            }
+        }
+
+        private async Task ValidateAsync(Feature feature)
```

Comment:
Split the checks for name and stub so it can return more specific error messages